src/hotspot/share/oops/instanceStackChunkKlass.inline.hpp

```diff
@@ -791,10 +791,16 @@ inline BitMap::idx_t InstanceStackChunkKlass::bit_offset(int stack_size_in_words
 template <bool mixed>
 void InstanceStackChunkKlass::run_nmethod_entry_barrier_if_needed(const StackChunkFrameStream<mixed>& f) {
   CodeBlob* cb = f.cb();
-  if ((mixed && cb == nullptr) || !cb->is_nmethod()) return;
-  nmethod* nm = cb->as_nmethod();
-  if (BarrierSet::barrier_set()->barrier_set_nmethod()->is_armed(nm)) {
-    nm->run_nmethod_entry_barrier();
+  if ((mixed && cb == nullptr) || !cb->is_nmethod()) {
```

Comment:
If `mixed` is false, then there are no interpreted frames. The condition should be `if (f.is_interpreted())` (which short-circuits to false if `mixed` is false).

Reply:
Ok that's much better.  Thanks.  If (!f.is_interpreted) then cb will always be an nmethod? Or do I need to check for that in the else case?

Reply:
see revised version.

Reply:
There's one other possibility: a safepoint stub (in the case of forced preemption). But `StackChunkFrameStream` also has `is_compiled` and `is_stub`. Exactly one of those three would be true for each frame. If `is_compiled`, then you have an nmethod.

Reply:
I changed this so that it checks if it's interpreted, and then in the else check if cb->is_nmethod() to exclude the stub case.  Does it look right?

Reply:
Yep. Looks good.

Reply:
Thanks!